core/domain/topic_services.py

```diff
@@ -315,11 +314,30 @@ def apply_change_list(topic_id, change_list):
                     raise Exception('Invalid change dict.')
             elif (change.cmd ==
                   subtopic_page_domain.CMD_UPDATE_SUBTOPIC_PAGE_PROPERTY):
+                subtopic_page_id = (
+                    subtopic_page_domain.SubtopicPage.get_subtopic_page_id(
+                        topic_id, change.id))
+                index = next(
+                    (index for index, subtopic_page in enumerate(
+                        modified_subtopic_pages)
+                     if subtopic_page.id == subtopic_page_id), None)
+                if index is None:
+                    subtopic_page = (
```

Comment:
Perhaps better to preprocess the changelist at the beginning for any subtopic page edits that have ID less than next_subtopic_page_id, and extract the affected subtopic html pages at the outset in a single get_multi() call. Then refer to them here.

Reply:
Done.